presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -387,4 +424,29 @@ public void setNull(PreparedStatement statement, int index)
             }
         };
     }
+
+    private static class DataCompressionDao
+    {
+        private final Handle handle;
+
+        private DataCompressionDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Optional<DataCompression> getDataCompression(JdbcTableHandle table)
+        {
+            return handle.createQuery("" +
+                    "SELECT data_compression_desc FROM sys.partitions p " +
+                    "INNER JOIN sys.tables t ON p.object_id = t.object_id " +
+                    "INNER JOIN sys.indexes i ON t.object_id = i.object_id " +
+                    "WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name AND i.type IN (0,1) " +
+                    "AND i.data_space_id NOT IN (SELECT data_space_id FROM sys.partition_schemes)")
+                    .bind("schema", table.getSchemaName())
+                    .bind("table_name", table.getTableName())
+                    .mapTo(String.class)
+                    .findOne()
```

Comment:
What if query returns more than one record, is it going to throw?

Reply:
Yes !! If it return 0 value it returns `Optional#empty` if more than 1 row it throws an exception